server/middleware/versions.coffee

```diff
@@ -89,7 +94,11 @@ module.exports =
 
     doc.set('parent', latest._id)
 
-    doc = yield doc.save()
+    try
+      doc = yield doc.save()
+    catch e
+      yield latest.update({$set: _.pick(latest.toObject(), 'version', 'index', 'slug')})
+      throw e
```

Comment:
And in general compensating for the unwanted side-effects of other endpoints feels a code smell, but I dunno how hard it would be to refactor such that that side effect only happens when desired.